lib/scenic/adapters/postgres.rb

```diff
@@ -155,17 +155,36 @@ def create_materialized_view(name, sql_definition, no_data: false)
       # @param no_data [Boolean] Default: false. Set to true to create
       #   materialized view without running the associated query. You will need
       #   to perform a non-concurrent refresh to populate with data.
+      # @param side_by_side [Boolean] Default: false. Set to true to create the
+      #   new version under a different name and atomically swap them, limiting
+      #   downtime at the cost of doubling disk usage
       #
       # @raise [MaterializedViewsNotSupportedError] if the version of Postgres
       #   in use does not support materialized views.
       #
       # @return [void]
-      def update_materialized_view(name, sql_definition, no_data: false)
+      def update_materialized_view(
+        name, sql_definition, no_data: false, side_by_side: false
+      )
         raise_unless_materialized_views_supported
 
-        IndexReapplication.new(connection: connection).on(name) do
-          drop_materialized_view(name)
-          create_materialized_view(name, sql_definition, no_data: no_data)
+        if side_by_side
+          session_id = Time.now.to_i
+          new_name = "#{name}_new_#{session_id}"
+          old_name = "#{name}_drop_#{session_id}"
+          IndexReapplication.new(connection: connection).on_side_by_side(
+            name, new_name, session_id
+          ) do
+            create_materialized_view(new_name, sql_definition, no_data: no_data)
+          end
+          rename_materialized_view(name, old_name)
```

Comment:
Is there a reason not to `drop` here?

Reply:
I think I did it this way out of the assumption that we might commit after the rename (but before the drop) and pause to give other threads a chance to finish using that view before we drop it, to reduce the lock impact. I didn't actually build that, of course.